pkg/operators/ebpf/converters.go

```diff
@@ -106,10 +112,66 @@ func (i *ebpfInstance) initEnumConverter(gadgetCtx operators.GadgetContext) erro
 	return nil
 }
 
+func (i *ebpfInstance) initStackConverter(gadgetCtx operators.GadgetContext) error {
+	kernelSymbolResolver, err := kallsyms.NewKAllSyms()
+	if err != nil {
+		return err
+	}
+	for _, ds := range gadgetCtx.GetDataSources() {
+		for _, in := range ds.GetFieldsWithTag(StackIdKernelType) {
+			if in == nil {
+				continue
+			}
+			in.SetHidden(true, false)
+			out, err := ds.AddField(in.Name()+"_symbolized", api.Kind_String)
+			if err != nil {
+				return err
+			}
+			converter := func(ds datasource.DataSource, data datasource.Data) error {
+				if i.stackIdMap == nil {
+					return errors.New("kernel stack map is not initialized but used. " +
+						"if you are using `gadget_kernel_stack` as event field, " +
+						"try to include <gadget/stack_map.h>")
```

Comment:
Should this error be returned in the `converter` function or in the parent function `initStackConverter`?

cc @flyth

Reply:
I can move the check to `initStackConverter`. And it needs the kernel stack map to be initialized in `init` instead of `Start`.